### PR TITLE
docs(rxPageTitle): FRMW-187-Migrate rxPageTitle to API Docs

### PR DESCRIPTION
**Pull request by 100stacks**
JIRA: https://jira.rax.io/browse/FRMW-187

- [x] Dev LGTM
- [x] Dev LGTM
- [x] Dev LGTM

Migrate `rxPageTitle` to API Documentation.

### CODE REVIEW

**Comment by bry4n** [comment]
LGTM

**Comment by ssalinas24** [comment]
LGTM

File: src/rxPageTitle/README.md [hunk 1]
```diff
@@ -1,14 +1,3 @@
...
-
-The second is `setTitleUnsafeStripHTML()`. This will strip any HTML tags from the string, and set the title to the result. This uses
-the [technique found here](http://stackoverflow.com/questions/5002111/javascript-how-to-strip-html-tags-from-string). Note the caveats
-listed there, namely:
-
-  1. Only tags valid within `<div>` will be correctly stripped out
-  2. You should not have `<script>` within the title
-  3. You should not pass `null` as the title
-  4. The title must come from a trusted source, i.e. danger danger danger `<img onerror='alert(\"could run arbitrary JS here\")' src=bogus>`
+`rxPageTitle` component manages page titles.
```
Comment (CITguy) [inline src/rxPageTitle/README.md]: I would match this with the API overview description (no backticks).

File: src/rxPageTitle/rxPageTitle.js [hunk 2]
```diff
@@ -16,7 +16,22 @@ angular.module('encore.ui.rxPageTitle', [])
 /**
  * @ngdoc service
  * @name rxPageTitle.service:rxPageTitle
- * @description [TBD]
+ * @description
+ *  `rxPageTitle` service manages page titles.
+ *
+ * Two methods are available for setting the page title: 
+ * 1. First, `setTitle()` simply sets the title to whatever raw string is passed in.
```
Comment (CITguy) [inline src/rxPageTitle/rxPageTitle.js]: Provide a new line above this to trigger a markdown numbered list.

File: src/rxPageTitle/rxPageTitle.js [hunk 3]
```diff
@@ -16,7 +16,22 @@ angular.module('encore.ui.rxPageTitle', [])
...
+ * 2. Second, `setTitleUnsafeStripHTML()` strips any HTML tags from the string, and sets the title to 
+ * the result. This uses the 
+ * [technique found here](http://stackoverflow.com/questions/5002111/javascript-how-to-strip-html-tags-from-string). 
+ * Note the caveats listed there, namely:
+ * 
+ *   * Only tags valid within `<div>` will be correctly stripped out
+ *   * You should not have  any `<script>` tags within the title
+ *   * You should not pass `null` as the title
+ *   * The title must come from a trusted source, i.e. danger danger danger 
+ *   `<img onerror='alert(\"could run arbitrary JS here\")' src=bogus>`
```
Comment (CITguy) [inline src/rxPageTitle/rxPageTitle.js]: * swap the single and double quotes to eliminate the escaped quotes
* use self-closing ``

```
`<img onerror="alert('could run arbitrary JS here')" src=bogus />`
```

File: src/rxPageTitle/rxPageTitle.js [hunk 4]
```diff
@@ -67,10 +82,10 @@ angular.module('encore.ui.rxPageTitle', [])
  * Given a string, it removes all HTML tags from the string, using the
  * browser's own parsing engine. Any content inside of tags will be kept.
  *
- * NOTE: You must only use this with trusted text. See
- * http://stackoverflow.com/a/5002618 for more details
+ * **NOTE:** You must only use this with **trusted text**. See this
+ * {@link http://stackoverflow.com/a/5002618 StackOverflow} answer for more details.
  *
- * @param {String} The string to remove HTML from
+ * @param {String} $document The string to remove HTML from **trusted text**
```
Comment (CITguy) [inline src/rxPageTitle/rxPageTitle.js]: `$document` is the incorrect name for the param (it is the dependency injected to the filter). The correct name is `htmlString`. Please update.

File: src/rxPageTitle/rxPageTitle.js (hunk 2 above)
Comment (100stacks) [inline src/rxPageTitle/rxPageTitle.js]: @CITguy thanks for the tip :+1:

**Comment by 100stacks** [comment]
Okay comments incorporated. :+1:

File: src/rxPageTitle/rxPageTitle.js [hunk 5]
```diff
@@ -16,7 +17,23 @@ angular.module('encore.ui.rxPageTitle', [])
...
+ * [technique found here](http://stackoverflow.com/questions/5002111/javascript-how-to-strip-html-tags-from-string). 
+ * Note the caveats listed there, namely:
+ * 
+ *   * Only tags valid within `<div>` will be correctly stripped out
+ *   * You should not have  any `<script>` tags within the title
+ *   * You should not pass `null` as the title
+ *   * The title must come from a trusted source, i.e. danger danger danger 
+ *   `<img onerror="alert('could run arbitrary JS here')" src=bogus />`
+ *   
  */
```
Comment (CITguy) [inline src/rxPageTitle/rxPageTitle.js]: To adhere to what was originally in the README, please update this comment to the following:

```
/**
 * @ngdoc service
 * @name rxPageTitle.service:rxPageTitle
 * @description
 *  `rxPageTitle` service manages page titles.
 *
 * Two methods are available for setting the page title:
 *
 * First, `setTitle()` simply sets the title to whatever raw string is passed in.
 *
 * Second, `setTitleUnsafeStripHTML()` strips any HTML tags from the string, and sets the title to
 * the result. This uses the
 * [technique found here](http://stackoverflow.com/questions/5002111/javascript-how-to-strip-html-tags-from-string).
 * Note the caveats listed there, namely:
 *
 * 1. Only tags valid within `<div>` will be correctly stripped out
 * 2. You should not have  any `<script>` tags within the title
 * 3. You should not pass `null` as the title
 * 4. The title must come from a trusted source, i.e. danger danger danger
 *    `<img onerror="alert('could run arbitrary JS here')" src=bogus />`
 */
```

## Current
screen shot 2015-10-05 at 4 06 08 pm

## API Documentation
Generated from comment block above.
screen shot 2015-10-05 at 4 06 17 pm

**Comment by 100stacks** [comment]
@CITguy okay updated formatting. :+1:

**Comment by CITguy** [comment]
LGTM

@100stacks, please squash PR

**Comment by 100stacks** [comment]
@CITguy it's squashed :pushpin: :+1:

**Pull request merged**